pypesto/ensemble/ensemble.py

```diff
@@ -192,14 +205,31 @@ def _stack_outputs_sensi(ic: int):
             # stack into one numpy array
             return np.stack(output_sensi_list, axis=-1)
 
-        def _compute_summary(tmp_array, percentiles_list):
+        def _stack_weights(ic: int):
+            """
+            Group weights for different parameter vectors of one ensemble
+            together, if they belong to the same simulation condition, and
+            stacks them in one array
+            """
+            # Were outputs computed
+            if self.prediction_results[0].conditions[ic].output is None:
+                return None
+            # stack predictions
+            output_list = [prediction.conditions[ic].output[AMICI_LLH]
+                           for prediction in self.prediction_results]
+            # stack into one numpy array
+            return np.stack(output_list, axis=-1)
+
+        def _compute_summary(tmp_array, percentiles_list, weights):
             """
             Computes means, standard deviation, median, and requested
             percentiles for a set of stacked simulations
             """
             summary = {}
-            summary[MEAN] = np.mean(tmp_array, axis=-1)
-            summary[STANDARD_DEVIATION] = np.std(tmp_array, axis=-1)
+            summary[MEAN] = np.average(tmp_array, axis=-1, weights=weights)
+            summary[STANDARD_DEVIATION] = np.average(
+                (tmp_array.T-summary[MEAN].T).T**2,
+                axis=-1, weights=weights)
```

Comment:
Variance?

Reply:
oh yes, there needs to be an additional np.sqrt() 👍

Reply:
Then equivalent to `np.average(np.abs(tmp_array - summary[MEAN]), axis=-1, weights=weights)` I guess

Reply:
no, at least that should not be the case mathematically...